sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/newparser/checks/NewXmlVerifier.java

```diff
@@ -84,15 +96,44 @@ private static NewXmlVerifier createVerifier(String fileName, NewXmlCheck check)
 
     context.fileSystem().add(defaultInputFile);
 
-    NewXmlFile newXmlFile;
+    NewXmlFile xmlFile;
     try {
-      newXmlFile = NewXmlFile.create(defaultInputFile);
+      xmlFile = NewXmlFile.create(defaultInputFile);
     } catch (IOException e) {
       throw new IllegalStateException(String.format("Unable to scan xml file %s", filePath), e);
     }
 
-    check.scanFile(context, newXmlFile);
-    return new NewXmlVerifier(context.allIssues());
+    check.scanFile(context, xmlFile);
+    return new NewXmlVerifier(xmlFile, context.allIssues());
+  }
+
+  private void checkIssues() {
+    SingleFileVerifier fileVerifier = SingleFileVerifier.create(file.getInputFile().path(), StandardCharsets.UTF_8);
+    addComments(fileVerifier, file.getDocument());
+
+    issues.forEach(issue -> {
+      IssueLocation loc = issue.primaryLocation();
+      TextRange textRange = loc.textRange();
+      fileVerifier
+        .reportIssue(loc.message())
+        .onRange(
+          textRange.start().line(),
+          textRange.start().lineOffset() + 1,
+          textRange.end().line(),
+          textRange.end().lineOffset() + 1);
+    });
+
+    fileVerifier.assertOneOrMoreIssues();
+  }
+
+  private void addComments(SingleFileVerifier fileVerifier, Node node) {
+    if (node.getNodeType() == Node.COMMENT_NODE) {
+      Comment comment = (Comment) node;
+      XmlTextRange range = NewXmlFile.nodeLocation(node);
+      fileVerifier.addComment(range.getStartLine(), range.getStartColumn() + 1 + 4, comment.getNodeValue(), 0, 0);
```

Comment:
4? magic number? maybe use a constant for that, I guess it's `<!--`

Reply:
I will fix it my next PR